### PR TITLE
Other adults refactor for api

### PR DESCRIPTION
**Pull request by lauherkcambria**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- none - waiting for nesh's enzym pr ;)

#### Which User story does this relate to?
- #147282229

#### Brief feature description
- refactor to be more inline with other cards 

#### Does this PR block/effect other stories?
- nope

#### Do any commands needs to be run in order to run the project with this PR?
- nope

#### Code coverage percent diff as a result of this PR?
- no diff

#### JS errors diff as a result of this PR?
- no diff

### CODE REVIEW

**Comment by lauherkcambria** [comment]
jenkins, retest this please

File: app/javascript/forms/OtherAdultsCardsGroup.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import Immutable from 'immutable'
+import React from 'react'
+import PropTypes from 'prop-types'
+import {PhoneNumberField} from '../common/phoneNumberFields'
```
Comment (ratneshraval) [inline app/javascript/forms/OtherAdultsCardsGroup.js]: Are you using this?

File: app/javascript/forms/OtherAdultsCardsGroup.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import {getDictionaryId, checkArrayObjectPresence} from '../helpers/commonHelper.jsx'
+import {OtherAdultsCardField} from '../common/OtherAdultsCardField'
+
+const otherAdultsDefaults = Object.freeze({
+  otherAdultsComponents: [],
```
Comment (ratneshraval) [inline app/javascript/forms/OtherAdultsCardsGroup.js]: These 2 fields are not used, right?

File: app/javascript/forms/OtherAdultsCardsGroup.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      otherAdultsList = otherAdultsList.push(otherAdultsDefaults)
+    }
+    otherAdultsList = otherAdultsList.update(cardIndex, x => x.set(type, value))
+    this.props.setParentState('otherAdults', otherAdultsList.toJS())
+  }
+
+  handleNameFieldInput (index, value, type) {
+    let otherAdultsList = Immutable.fromJS(checkArrayObjectPresence(this.props.otherAdults) || [otherAdultsDefaults])
+    otherAdultsList = otherAdultsList.update(index,
+       otherAdult => otherAdult.setIn(['nameField', type], value))
```
Comment (ratneshraval) [inline app/javascript/forms/OtherAdultsCardsGroup.js]: 👍

File: app/javascript/forms/index.js [hunk 4]
```diff
@@ -111,12 +112,18 @@ export default class Forms extends React.Component {
...
 
             <div className='cards-section col-xs-12 col-sm-12 col-md-12 col-lg-12'>
               <h3>V. Applicant (S) - <span>Other Adults</span></h3>
-              <OtherAdultsCard {...this.props} />
+              <OtherAdultsCard
+                focusComponentName={this.state.focusComponentName}
+                setFocusState={this.setFocusState}
+                setParentState={this.setApplicationState}
+                otherAdults={this.state.application.otherAdults}
+                {...this.props} />
```
Comment (ratneshraval) [inline app/javascript/forms/index.js]: Sadness :(

File: app/javascript/common/OtherAdultsCardField.js [hunk 5]
```diff
@@ -3,106 +3,36 @@ import {InputComponent} from './inputFields'
...
-    })
-  }
-
   render () {
-    return (
+    const adult = this.props.otherAdults
 
+    return (
       <div className='row list-item' >
         <div > <span onClick={() => this.clickClose(this.props.id)} className='pull-right glyphicon glyphicon-remove' />
```
Comment (ratneshraval) [inline app/javascript/common/OtherAdultsCardField.js]: Please bring the clickClose span UI to a level up

File: app/javascript/forms/index.js (hunk 4 above)
Comment (lauherkcambria) [inline app/javascript/forms/index.js]: why?

File: app/javascript/common/OtherAdultsCardField.js (hunk 5 above)
Comment (lauherkcambria) [inline app/javascript/common/OtherAdultsCardField.js]: i'm not quite sure i follow. This element is in each card so should be in the individual card field levels.

File: app/javascript/forms/index.js (hunk 4 above)
Comment (ratneshraval) [inline app/javascript/forms/index.js]: {...this.props}

File: app/javascript/common/OtherAdultsCardField.js (hunk 5 above)
Comment (ratneshraval) [inline app/javascript/common/OtherAdultsCardField.js]: The otherAdultCardFields shouldn't care about closing, The otherAdultGroup should have the close UI and method, just like how it has add new card UI and method.

File: app/javascript/common/OtherAdultsCardField.js [hunk 6]
```diff
@@ -8,8 +8,7 @@ export class OtherAdultsCardField extends React.Component {
 
     return (
       <div className='row list-item' >
```
Comment (ratneshraval) [inline app/javascript/common/OtherAdultsCardField.js]: Please remove this div, it is adding extra height.

**Comment by lauherkcambria** [comment]
@ratneshraval updated 👍

**Pull request merged**